### ISSUE TITLE
Ignore variables in test packages/files

### ISSUE THREAD

**Issue by kaskavalci** [leighmcculloch/gochecknoglobals#11]
It is possible to initialize test variables by `TestMain`. However global variables are the only way to share data with `TestMain` and tests. Can we ignore global variables in `_test` packages or `_test.go` files in general?

**Comment by leighmcculloch** [comment]
That can cause side effects to leak from one test to the next though. Could you share an example of when a global variable is required for tests instead of a function that returns a setup to be shared across tests?

**Comment by GregoryVds** [comment]
Facing the same issue here. My setup is expensive to prepare (loading a BPF program into the kernel). I want to perform this setup step only once and then reuse it across several tests. So I basically get a file descriptor in TestMain that I want to share with my test cases.

**Comment by leonklingele** [comment]
`gochecknoglobals` lints tests only when a `-t` flag is provided. Why not run it without the flag?
Otherwise [(*testing.T).Run](https://golang.org/pkg/testing/#T.Run) might be of help.

**Comment by GregoryVds** [comment]
Thanks for the suggestion leon, I didn't know about this flag. I am actually using `golangci-lint `to run `gochecknoglobals` and it looks like it does not expose the `-t`. Anyway, I ended up using an exclude-rule to turn off gochecknoglobals on the single test file that was problematic.

**Comment by bombsimon** [comment]
A bit late to the party but `golangci-lint` has it's own flag for this but defaults to true. Have a look at `--tests=false` or `--skip-files=.*_tests`

**Comment by leighmcculloch** [comment]
It's also worth noting `golangci-lint` ports this linter into its code, so it does not actually call this `gochecknoglobals` implementation and it's implementation is based on an older version of this tool. There's an issue discussing that here https://github.com/golangci/golangci-lint/issues/1393.

**Comment by leighmcculloch** [comment]
I think the two recommendations above not using the `-t` flag, or using golangci-lint's `--tests` flag, solve the issue posted, so I am closing. Please reopen if this issue is not addressed or I am missing something.

**Comment by bbkane** [comment]
I'm a bit late to this party, but I'm using the "golden files" pattern ( https://speakerdeck.com/mitchellh/advanced-testing-with-go?slide=19 ) for my tests (see actual code [here](https://github.com/bbkane/warg/blob/8a92b9e5493805bd9354a6e85a283ff48dcb7892/app_help_ext_test.go#L22)). I really want to add `gochecknoglobals` for all code (tests included) in golangci-lint, but I'm not sure how to restructure my code to "unglobal" the `update` flag or have `gochecknoglobals` ignore this one specific flag in just this one test file. Any ideas? From this discussion it looks my options are either to:
- stop linting this file completely
- refactor my code (but I can't seem to see a way how)

Any ideas?

**Comment by leighmcculloch** [comment]
@bbkane There's a few options I'm aware of. Others might chime in with additional ideas. Here's what I know:

You can use `//nolint:gochecknoglobals`. (I haven't used this myself, so I think I have the syntax right, but you might need to fiddle. I've seen it before here: https://golangci-lint.run/usage/false-positives/).

Another option is to put that `var update = ` into its own file, and then stop linting that one file. You only need to put the definition into its own file, not the uses of the variable.

**Comment by bbkane** [comment]
That seems to work, thank you! 

```go
//nolint:gochecknoglobals  // https://github.com/leighmcculloch/gochecknoglobals/issues/11#issuecomment-1368578131
var update = stdlibflag.Bool("update", false, "update golden files")
```

Although if you (or anyone) hear about any techniques to write an `update` testing flag without using this package level variable trick, please update this issue.
